### ISSUE TITLE
[HELP] Problem accessing env vars in ts

### ISSUE THREAD

**Issue by ssb-jnk** [garronej/vite-envs#17]
Greetings!

I am having an issue using vite-envs. I have followed the starter guide to get everything up and running, and it does seem like things do happen behind the scenes, such as index.html including environment variables from --env in docker run. But when i try to access the variabels with `import.meta.env.A`, then its undefined or empty rather.

I have a server.js which runs ViteExpress and serves as the apps BFF, and then using nodemon to run it in prod.

```
"scripts": {
    "prepare": "npx vite-envs update-types",
    "dev": "nodemon server.js -w server.js",
    "prod": "NODE_ENV=production node server.js",
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "lint:fix": "eslint ./src --ext .jsx,.js,.ts,.tsx --quiet --fix --ignore-path ./.gitignore",
    "lint:format": "prettier  --loglevel warn --write \"./**/*.{js,jsx,ts,tsx,css,md,json}\" ",
    "preview": "vite preview"
  },
```
Vite-envs version
`"vite-envs": "^4.0.13"`

I have tried logging what import.meta.env contains, and it does contain the variables but not the values themselves, values are empty, even though the values are visible in index.html (i know this because i decoded the base64 to verify the values).

Variables are defined in .env (empty because i want to set it with docker):
```sh
A=
B=
C=
D=
```

Dockerfile:
```
FROM node:20-alpine AS builder
WORKDIR /usr/local/app
COPY . .
RUN npm ci && npm run build


FROM node:20-alpine
WORKDIR /usr/local/app

COPY --from=builder /usr/local/app/dist ./dist
COPY package*.json server.js ./

RUN npm install --save-exact express vite-express \
    && cp -r dist/* .

ENV PORT=8080
EXPOSE 8080

ENTRYPOINT ["sh", "-c", "./vite-envs.sh && npm run prod"]
```

I run it like this
```sh
docker run -it -p 8080:8080 \
  --env A="A" \
  --env B="B" \
  --env C="C" \
  --env D="D" local-app-test
```

Response in console when printing import.meta.env
```
{
    "BASE_URL": "/",
    "MODE": "production",
    "DEV": "false",
    "PROD": "true",
    "A": "",
    "B": "",
    "C": "",
    "D": ""
}
```

index.html in the docker container when app is running
```
<!DOCTYPE html><html lang="en"><head>
    <meta charset="UTF-8">
    <link rel="icon" type="image/svg+xml" href="/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>App test</title>

    <script data-script-description="Environment variables injected by vite-envs">
      var envWithValuesInBase64 = { CONTAINS ALL THE BASE64 KEY VALUE PAIRS }
      var env = {};
      Object.keys(envWithValuesInBase64).forEach(function (name) {
        env[name] = new TextDecoder().decode(
          Uint8Array.from(
            atob(envWithValuesInBase64[name]),
            c => c.charCodeAt(0))
        ).slice(0,-1);
      });
      window.__VITE_ENVS = env;
    </script><script type="module" crossorigin="" src="/assets/index-D6asdcv.js"></script>
    <link rel="stylesheet" crossorigin="" href="/assets/index-asdzcAwd.css">
  </head>
  <body>
    <div id="root"></div>


</body></html>
```

Do you have any idea to what might be wrong? Feels like i hit a plateau, so any help would be greatly appreciated

**Comment by ssb-jnk** [comment]
@garronej Thank you for responding so quickly!

I'll gladly accept any VIP treatment! The repository is set to internal, but will at some point be public, there is not any sensitive information in the repository. I have given you read access to the repository. [Link](https://github.com/statisticsnorway/dapla-ctrl/tree/vite-envs-not-injected) directly to the branch containing everything related to vite-envs. Any help is greatly appreciated, thanks!

Could also set up a call if that would be easier.

**Comment by garronej** [comment]
@ssb-jnk,  

Here you go:  

`Dockerfile`
```diff
 RUN npm install --save-exact express vite-express
-    && cp -r dist/* .

 ENV PORT=8080
 EXPOSE 8080

-ENTRYPOINT ["sh", "-c", "./vite-envs.sh && npm run prod"]
+ENTRYPOINT ["sh", "-c", "./dist/vite-envs.sh && npm run prod"]
```  

Best

**Comment by ssb-jnk** [comment]
Thank you for the response @garronej! And yes this solved the problem. I tried it earlier but i got a token error where it pointed to the data in the base64 array, but your response made me look more into it! It appears having a value like this "https://my-org.atlassian.net/wiki/x/EycA4g" (without quotes) causes invalid token error. I removed everything after .net, and then it worked. A little odd that it causes invalid token, but ill find a way around it.

Thank you so much! :)
